up42/__init__.py

```diff
@@ -57,10 +57,10 @@
         initialize_order,
         initialize_asset,
         initialize_webhook,
-        authenticate,
         get_webhooks,
         create_webhook,
         get_webhook_events,
         get_credits_balance,
     ]
+    if hasattr(obj, "__name__")
```

Comment:
Pylint is triggering an error if we do not add this. Alternative option is to set an ignore. :)

Reply:
It means something is off, otherwise we would get the warning before as well

Reply:
After some research, I finally found what is off here.

We have changed the `authenticate` variable to store the method of a Class. It seems that before `mypy` was able to infer that there were Classes, and functions in the list. Now, we have a third type that is variable, therefore we have in that list Classes, Functions, and one variable that store a method of a Class.

Since the list contains a mix of classes, functions, and variables, `mypy` infers the most general type that can accommodate all elements, which is likely `object`. The object type in Python doesn't have a __name__ attribute, leading to the error.

I figure out that splitting the assingment for Classes, and functions makes `mypy` happy and infers correctly the type. I also tested this and it works as well: 

```
__all__ = [
    Asset.__name__,
    Auth.__name__,
    Catalog.__name__,
    Order.__name__,
    Storage.__name__,
    Tasking.__name__,
    Webhook.__name__,
    get_example_aoi.__name__,
    read_vector_file.__name__,
    initialize_catalog.__name__,
    initialize_tasking.__name__,
    initialize_storage.__name__,
    initialize_order.__name__,
    initialize_asset.__name__,
    initialize_webhook.__name__,
    authenticate.__name__,
    get_webhooks.__name__,
    create_webhook.__name__,
    get_webhook_events.__name__,
    get_credits_balance.__name__,
] 
```
We can think on a better solution, perharps assinging `authenticate` to a better type or typing better, I did some test but I ended with splitting the assignment of the __all__ property to Classes and functions.